packages/client/src/encryption/SubscriberKeyExchange.ts

```diff
@@ -148,4 +149,13 @@ export class SubscriberKeyExchange {
             }
         }
     }
+
+    private async isAssignedToMe(streamPartId: StreamPartID, recipient: EthereumAddress, requestId: string): Promise<boolean> {
+        if (this.pendingRequests.has(requestId)) {
+            const erc1271Contract = this.subscriber.getERC1271ContractAddress(streamPartId)
+            const authenticatedUser = await this.authentication.getAddress()
+            return recipient === authenticatedUser || (recipient === erc1271Contract && erc1271Contract !== undefined)
+        }
```

Comment:
Can simplify the if statement as the `recipient` is never `undefined`

```suggestion
        if (this.pendingRequests.has(requestId)) {
            const authenticatedUser = await this.authentication.getAddress()
            const erc1271Contract = this.subscriber.getERC1271ContractAddress(streamPartId)
            return (recipient === authenticatedUser) || (recipient === erc1271Contract)
        }
```